src/AvPortalClientFactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_avportal;
+
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+
+/**
+ * Factory service to instantiate AV Portal clients.
+ */
+class AvPortalClientFactory implements AvPortalClientFactoryInterface, ContainerAwareInterface {
+
+  use ContainerAwareTrait;
+
+  /**
+   * Returns a new client instance.
```

Comment:
{@inheritdoc}

Reply:
Not really, I added the extra configuration parameters that can be passed to this factory. Different factories might have different parameters.